core/modules/user/user.module

```diff
@@ -3024,9 +3024,10 @@ function user_modules_uninstalled($modules) {
  * proper destination after a user has been properly logged in.
  */
 function user_login_destination() {
+  global $user;
   $destination = backdrop_get_destination();
   if ($destination['destination'] == 'user/login') {
-    $destination['destination'] = 'user';
+    $destination['destination'] = 'user/' . $user->uid;
```

Comment:
Maybe check for `$user->uid` first ?

Reply:
>Third-party authentication modules may use this function to determine the proper destination after a user has been properly logged in. 

After you've been "properly logged in", `$user->uid` should already exist.

Reply:
Fair enough